ffi/src/lib.rs

```diff
@@ -961,6 +962,19 @@ fn visit_expression_binary(
     }
 }
 
+fn visit_expression_unary(
+    state: &mut KernelExpressionVisitorState,
+    op: UnaryOperator,
+    inner_expr: usize,
+) -> usize {
+    if let Some(expr) = unwrap_kernel_expression(state, inner_expr).map(Box::new) {
+        wrap_expression(state, Expression::UnaryOperation { op, expr })
```

Comment:
Seems like we should define `Expression::unary` that takes `impl Into<Expression>`, similar to the existing `Expression::binary`? Then this whole method simplifies to:
```rust
unwrap_kernel_expression(state, inner_expr)
    .map_or_else(0, |expr| wrap_expression(state, Expression::unary(op, expr))
```

Reply:
Yep, that's nicer. We actually already had `Expression::unary` :)